tests/functional/test_paginator_config.py

```diff
@@ -0,0 +1,155 @@
+
+
+def _validate_known_pagination_keys(page_config):
+    for key in page_config:
+        if key not in KNOWN_PAGE_KEYS:
+            raise AssertionError("Unknown key '%s' in pagination config: %s"
+                                 % (key, page_config))
+
+
+def _valiate_result_key_exists(page_config):
+    if 'result_key' not in page_config:
+        raise AssertionError("Required key 'result_key' is missing "
+                             "from pagination config: %s" % page_config)
+
+
+def _validate_referenced_operation_exists(operation_name, service_model):
+    if operation_name not in service_model.operation_names:
+        raise AssertionError("Pagination config refers to operation that "
+                             "does not exist: %s" % operation_name)
+
+
+def _validate_operation_has_output(operation_name, service_model):
+    op_model = service_model.operation_model(operation_name)
+    output = op_model.output_shape
+    if output is None or not output.members:
+        raise AssertionError("Pagination config refers to operation "
+                             "that does not have any output: %s"
+                             % operation_name)
+
+
+def _validate_input_keys_match(operation_name, page_config, service_model):
+    input_tokens = page_config['input_token']
+    if not isinstance(input_tokens, list):
+        input_tokens = [input_tokens]
+    valid_input_names = service_model.operation_model(
+        operation_name).input_shape.members
+    for token in input_tokens:
+        if token not in valid_input_names:
+            raise AssertionError("input_token '%s' refers to a non existent "
+                                 "input member for operation: %s"
+                                 % (token, operation_name))
+    if 'limit_key' in page_config:
+        limit_key = page_config['limit_key']
+        if limit_key not in valid_input_names:
+            raise AssertionError("limit_key '%s' refers to a non existent "
+                                 "input member for operation: %s"
+                                 % (limit_key, operation_name))
+
+
+def _validate_output_keys_match(operation_name, page_config, service_model):
+    # NOTE: The original version of this function from translate.py had logic
+    # to ensure that the entire set of output_members was accounted for in the
+    # union of 'result_key', 'output_token', 'more_results', and
+    # 'non_aggregate_keys'.
+    # There's enough state drift (especially with non_aggregate_keys) that
+    # this is no longer a realistic thing to check.  Someone would have to
+    # backport the missing keys to all the paginators.
+    output_shape = service_model.operation_model(operation_name).output_shape
+    output_members = output_shape.members
+    for key_name, output_key in _get_all_page_output_keys(page_config):
+        if _looks_like_jmespath(output_key):
+            _validate_jmespath_compiles(output_key)
```

Comment:
One idea that may be interesting to pursue but may be a bit of work to do is actually try to validate the jmespath expression against the model. I could imagine using the [ArgumentGenerator](https://github.com/boto/botocore/blob/develop/botocore/utils.py#L502) and trying to apply the jmespath expression to that to see if you can get a match.

This would be super helpful if the PR actually has a jmespath expression because sure even though it is valid, I still will need to look into the model to see that the expression match up to the shape.

What's your thought on this? I do not think we need need it now but may be helpful in the future if we see a lot of paginators needing nested jmespath expressions.